### PR TITLE
test pmd rules

### PR DESCRIPTION
**Pull request by liorhar**

### CODE REVIEW

File: java/com/kenshoo/flume/rabbitmq/RabbitMqSink.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.applet.Applet;
```
Comment (bronson-io) [inline java/com/kenshoo/flume/rabbitmq/RabbitMqSink.java:28]: Avoid unused imports such as 'java.applet.Applet'

File: java/com/kenshoo/flume/rabbitmq/RabbitMqSink.java [hunk 2]
```diff
@@ -67,6 +68,11 @@ private static SinkBuilder builder() {
         return new RabbitMqSinkBuilder();
     }
 
+    @Override
+    public int hashCode(){
```
Comment (bronson-io) [inline java/com/kenshoo/flume/rabbitmq/RabbitMqSink.java:72]: Ensure you override both equals() and hashCode()

**Comment by liorhar** [comment]
@talsalmona @shlomisut 

bronson.io ... looks neat isn't it? free for open source, 79$ a month unlimited private repos. let's try?

**Comment by talsalmona** [comment]
+1

**Comment by shlomisut** [comment]
Looks neat, but I'm missing the "What rules do we have now"
this example is very simplistic

_[ Shlomi Sutton ]_
Software Architect, SEM // _Kenshoo_
_Office_ +972 (3) 746-6500 _570 // *Mobile_ +972 52 235 9949

---

_www.Kenshoo.com_ http://kenshoo.com/ - _We're Hiring!_
http://www.kenshoo.com/Careers_with_Kenshoo

On Mon, Nov 3, 2014 at 8:31 PM, Tal Salmona notifications@github.com
wrote:

> +1
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kenshoo/flume-rabbitmq-sink/pull/3#issuecomment-61524484
> .

## 

This e-mail, as well as any attached document, may contain material which 
is confidential and privileged and may include trademark, copyright and 
other intellectual property rights that are proprietary to Kenshoo Ltd, 
 its subsidiaries or affiliates ("Kenshoo"). This e-mail and its 
attachments may be read, copied and used only by the addressee for the 
purpose(s) for which it was disclosed herein. If you have received it in 
error, please destroy the message and any attachment, and contact us 
immediately. If you are not the intended recipient, be aware that any 
review, reliance, disclosure, copying, distribution or use of the contents 
of this message without Kenshoo's express permission is strictly prohibited.

**Comment by liorhar** [comment]
log into bronson.io ane check out the rules dashboard, this is basically all design rules of pmd
http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html

**Comment by liorhar** [comment]
sorry, not only design rules, also finalizers, exceptions, braces, optimizations, exceptions, string and more
http://pmd.sourceforge.net/pmd-4.3.0/rules/finalizers.html
http://pmd.sourceforge.net/pmd-4.3.0/rules/braces.html
http://pmd.sourceforge.net/pmd-4.3.0/rules/optimizations.html

**Comment by shlomisut** [comment]
basically - Yalla, why not. worth a shot.
practically - we should inform/consult the TL forum before we do this.

_[ Shlomi Sutton ]_
Software Architect, SEM // _Kenshoo_
_Office_ +972 (3) 746-6500 _570 // *Mobile_ +972 52 235 9949

---

_www.Kenshoo.com_ http://kenshoo.com/ - _We're Hiring!_
http://www.kenshoo.com/Careers_with_Kenshoo

On Mon, Nov 3, 2014 at 8:56 PM, Lior Harel notifications@github.com wrote:

> sorry, not only design rules, also finalizers, exceptions, braces,
> optimizations, exceptions, string and more
> http://pmd.sourceforge.net/pmd-4.3.0/rules/finalizers.html
> http://pmd.sourceforge.net/pmd-4.3.0/rules/braces.html
> http://pmd.sourceforge.net/pmd-4.3.0/rules/optimizations.html
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kenshoo/flume-rabbitmq-sink/pull/3#issuecomment-61528342
> .

## 

This e-mail, as well as any attached document, may contain material which 
is confidential and privileged and may include trademark, copyright and 
other intellectual property rights that are proprietary to Kenshoo Ltd, 
 its subsidiaries or affiliates ("Kenshoo"). This e-mail and its 
attachments may be read, copied and used only by the addressee for the 
purpose(s) for which it was disclosed herein. If you have received it in 
error, please destroy the message and any attachment, and contact us 
immediately. If you are not the intended recipient, be aware that any 
review, reliance, disclosure, copying, distribution or use of the contents 
of this message without Kenshoo's express permission is strictly prohibited.

**Pull request closed without merging**